dependencies.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# $1 MUST be 'init' or 'update'.
+
+
+# get submodules
+SM_CONFIG=$(cat submodules/ocrd_all/.gitmodules)
+
+SUBMODULE_NAMES=$(
+    for LINE in $SM_CONFIG
+    do
+        echo $LINE | grep -oE '"(.+?)"' | cut -d'"' -f2
+    done
+)
+
+CURRENT_DIR=$PWD
+
+# get the main OCR-D dependencies which stem from core and that most processors have in common.
+cd 'submodules/ocrd_all/core' || exit
+python -m venv venv
+VENV=$PWD'/venv/bin/python'
+CMD='-m pip install ocrd'
+$VENV $CMD
+CMD='-m pip freeze -l'
+$VENV $CMD > $CURRENT_DIR'/core_deps.txt'
+cd $CURRENT_DIR || exit
+
+echo '{' >> $CURRENT_DIR'/deps.json'
+
+for NAME in $SUBMODULE_NAMES
+do
+    if [ $NAME != 'tesseract' ] && [ $NAME != 'opencv-python' ]; then
+        DIR='/submodules/ocrd_all/'$NAME
+        echo "Currently processing " $NAME " ..."
+        cd $CURRENT_DIR$DIR || exit
+        python -m venv venv
+        VENV=$PWD'/venv/bin/python'
+
+        # install pkgs
+        if [ -f 'requirements.txt' ]; then
+            CMD_INSTALL=$(echo '-m pip install -r requirements.txt .')
+            $VENV $CMD_INSTALL
+        elif [ -f 'setup.py' ]; then
+            CMD_INSTALL=$(echo '-m pip install .')
+            $VENV $CMD_INSTALL
+        fi
+
+        # get deps and save them to JSON
+        CMD_FREZE=$(echo '-m pip freeze -l')
```

Comment:
The echo/shell callout seems redundant.
```suggestion
        CMD_FREZE='-m pip freeze -l'
```